pinot-core/src/main/java/org/apache/pinot/core/segment/index/loader/invertedindex/TextIndexHandler.java

```diff
@@ -134,6 +136,53 @@ private void checkUnsupportedOperationsForTextIndex(ColumnMetadata columnMetadat
     }
   }
 
+  private void createFSTIndexForColumn(ColumnMetadata columnMetadata) throws IOException {
```

Comment:
This should be enough for FST index creation. We shouldn't change createTextIndexForColumn() method as it will break the functionality

Reply:
On the same note, we should model this as a different text index type. Check TextIndexType enum. Currently it has Lucene.
May be we can have LUCENE_FST as another type.

Reply:
Yup, I am calling it just "FST" in the FieldConfig  (https://github.com/pradeepgv42/incubator-pinot/pull/2/files#diff-6d51e3f41e478650e105a6b5960bd4f6) 
or are you referring to something else?

Reply:
Yes, new index type in FieldConfig is good.